### PR TITLE
Add shading.

### PR DESCRIPTION
**Pull request by htmldoug**
## Summary
Does the stuff in https://github.com/rallyhealth/sbt-shading. Porting the plugin to mill is out of scope for now.

I've also deleted the legacy/deprecated stuff.

### CODE REVIEW

File: bench/src-js/Main.scala [hunk 1]
```diff
@@ -15,21 +15,15 @@ object Main{
       Common.playJson(duration)
       Common.circe(duration)
       Common.upickleDefault(duration)
-      Common.upickleLegacy(duration)
```
[the remark this answers is not in the document]
Comment (htmldoug) [reply bench/src-js/Main.scala]: Yep.

File: bench/src-jvm/Main.scala [hunk 2]
```diff
@@ -22,7 +22,7 @@ object Main{
       println("RUN JVM: " + duration)
       println()
 
-      Main.ujsonAst(duration)
+      Main.upickleAst(duration)
```
[the remark this answers is not in the document]
Comment (htmldoug) [reply bench/src-jvm/Main.scala]: Probably a rename mistake.

File: ujson/src/com/rallyhealth/ujson/v1/AsyncParser.scala [hunk 3]
```diff
@@ -58,17 +58,17 @@ object AsyncParser {
...
-                                            protected[ujson] var curr: Int,
-                                            protected[ujson] var stack: List[ObjArrVisitor[_, J]],
-                                            protected[ujson] var path: List[Any],
-                                            protected[ujson] var data: Array[Byte],
-                                            protected[ujson] var len: Int,
-                                            protected[ujson] var allocated: Int,
-                                            protected[ujson] var offset: Int,
-                                            protected[ujson] var done: Boolean,
-                                            protected[ujson] var streamMode: Int
+final class AsyncParser[J] protected[v1](
```
[the remark this answers is not in the document]
Comment (htmldoug) [reply ujson/src/com/rallyhealth/ujson/v1/AsyncParser.scala]: Same package structure we use internally, e.g. com/rallyhealth/rq/v1/RqClient.scala

**Comment by htmldoug** [comment]
> was there any sort of decent tooling help to deal with this mess, or did you have to do it by hand?

```
sed -e 's/ujson/com.rallyhealth.ujson.v1/g' -i '' **/*.scala
sed -e 's/upack/com.rallyhealth.upack.v1/g' -i '' **/*.scala
sed -e 's/upickle/com.rallyhealth.upickle.v1/g' -i '' **/*.scala
sed -e 's/[[]com.rallyhealth.ujson.v1[]]/[v1]/g' -i '' **/*.scala
sed -e 's/[[]com.rallyhealth.upickle.v1[]]/[v1]/g' -i '' **/*.scala

for FILE in $(find . -name '*.scala' | egrep -v 'out|target'); do
    NEW_FILE=$FILE
    for TERM in upickle upack ujson; do
        NEW_FILE=$(echo $NEW_FILE | gsed -r "s|(.*)$TERM(.*?)|\1com/rallyhealth/$TERM/v1\2|") # replace last match
    done

    mkdir -p $(dirname $NEW_FILE)

    mv -v $FILE $NEW_FILE
done
```

That took care of ~90%. I did the rest of the cleanup by hand.

File: core/src/com/rallyhealth/upickle/v1/core/Types.scala [hunk 4]
```diff
@@ -300,4 +300,4 @@ trait Types{ types =>
     }
   }
 
-}
\ No newline at end of file
+}
```
Comment (usufruct99) [inline core/src/com/rallyhealth/upickle/v1/core/Types.scala]: unacceptable

File: upickle/test/src/com/rallyhealth/upickle/v1/FailureTests.scala [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+/**
+* Generally, every failure should be a Invalid.Json or a
+* InvalidData. If any assertion errors, match errors, number
+* format errors or similar leak through, we've failed
+*/
+object FailureTests extends TestSuite {
+
+  def tests = Tests {
+//    test("test"){
+//      read[com.rallyhealth.ujson.v1.Value](""" {unquoted_key: "keys must be quoted"} """)
```
Comment (usufruct99) [inline upickle/test/src/com/rallyhealth/upickle/v1/FailureTests.scala]: why is this commented out?

File: upickle/test/src/com/rallyhealth/upickle/v1/FailureTests.scala [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+//        """ "A JSON payload should be an object or array, not a string." """,
+        """ {"Extra value after close": true} "misplaced quoted value" """,
+        """ {"Illegal expression": 1 + 2} """,
+        """ {"Illegal invocation": alert()} """,
+        """ {"Numbers cannot have leading zeroes": 013} """,
+        """ {"Numbers cannot be hex": 0x14} """,
+        """ ["Illegal backslash escape: \x15"] """,
+        """ [\naked] """,
+        """ ["Illegal backslash escape: \017"] """,
+//        """ [[[[[[[[[[[[[[[[[[[["Too deep"]]]]]]]]]]]]]]]]]]]] """,
```
Comment (usufruct99) [inline upickle/test/src/com/rallyhealth/upickle/v1/FailureTests.scala]: image

File: upickle/test/src/com/rallyhealth/upickle/v1/FailureTests.scala (hunk 5 above)
Comment (htmldoug) [reply upickle/test/src/com/rallyhealth/upickle/v1/FailureTests.scala]: You'd have to git blame upstream. I didn't write this.

**Pull request merged**